packages/package-manager/src/CocoaPodsPackageManager.ts

```diff
@@ -315,6 +346,22 @@ export class CocoaPodsPackageManager implements PackageManager {
     }
   }
 
+  // Exposed for testing
+  spawnPodCommandAsync(
+    args?: readonly string[],
+    options?: SpawnOptions
+  ): SpawnPromise<SpawnResult> {
+    if (this.shouldUseRubyBundler) {
```

Comment:
This makes me think that RubyBundler should be some sort of subclass or different class. If you need to pass in a boolean that changes core functionality of most features then it is basically a different class.

Reply:
Hmm, I maybe can do that (Still new to TS), but all this is really doing is changing executions from `pod <args>` to `bundle exec pod <args>`. At the command line level it's super simple.

Reply:
I finally got back to this. Removed the boolean flag, create a subclass, and created a "factory" static `create` method for returning the appropriate instance.